Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -148,109 +218,177 @@ class LoginDetailViewController: UITableViewController {
+    func textFieldShouldReturn(textField: UITextField) -> Bool {
+        let usernameField = textFieldForItem(.UsernameItem)
+        let passwordField = textFieldForItem(.PasswordItem)
+        let websiteField = textFieldForItem(.WebsiteItem)
+
+        if textField == usernameField {
+            passwordField?.becomeFirstResponder()
+        } else if textField == passwordField {
+            websiteField?.becomeFirstResponder()
+        }
+
         return false
     }
+
+    func textFieldShouldBeginEditing(textField: UITextField) -> Bool {
+        if let passwordField = textFieldForItem(.PasswordItem) where passwordField == textField {
+            cellForItem(.PasswordItem)?.displayDescriptionAsPassword = false
+        }
+        return true
+    }
+
+    func textFieldDidEndEditing(textField: UITextField) {
+        if let passwordField = textFieldForItem(.PasswordItem) where passwordField == textField {
+            cellForItem(.PasswordItem)?.displayDescriptionAsPassword = true
+        }
+    }
 }
 
 // MARK: - Selectors
 extension LoginDetailViewController {
 
+    func SELonProfileDidFinishSyncing() {
+        // Reload details after syncing.
+        profile.logins.getLoginDataForGUID(login.guid).uponQueue(dispatch_get_main_queue()) { result in
+            if let syncedLogin = result.successValue {
+                self.login = syncedLogin
+            }
+        }
+    }
+
     func SELedit() {
-        tableView.editing = true
+        editingInfo = true
+
+        let cell = tableView.cellForRowAtIndexPath(InfoItem.UsernameItem.indexPath) as! LoginTableViewCell
+        cell.descriptionLabel.becomeFirstResponder()
+
         navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .Done, target: self, action: "SELdoneEditing")
     }
 
     func SELdoneEditing() {
-        tableView.editing = false
+        let usernameField = textFieldForItem(.UsernameItem)
+        let passwordField = textFieldForItem(.PasswordItem)
+        let websiteField = textFieldForItem(.WebsiteItem)
+
+        // Force resign responder from all input fields
+        [usernameField, passwordField, websiteField].forEach { $0?.resignFirstResponder() }
+
+        editingInfo = false
         navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .Edit, target: self, action: "SELedit")
+
+        let username = usernameField?.text ?? ""
+        let password = passwordField?.text ?? ""
+        let hostname = websiteField?.text ?? ""
+
+        // Update login if there were any changes
+        if username != login.username || password != login.password || hostname != login.hostname {
+            // Update local, in-memory copy to view changes right away.
+            login = Login(guid: login.guid, hostname: hostname, username: username, password: password)
+            profile.logins.updateLoginByGUID(login.guid, new: login, significant: true)
```

Comment:
Two options: 

1. Show an error prompt indicating that the update failed. This would require new strings.
2. Silently fail.

Ideally option (1) would be better if we're not 'string-frozen'.